frontend/src/metabase/modes/components/drill/ActionClickDrill/ActionClickDrill.tsx

```diff
@@ -0,0 +1,43 @@
+import { getDataFromClicked } from "metabase/lib/click-behavior";
+
+import { executeRowAction } from "metabase/dashboard/actions";
+
+import type { ParameterMappedForActionExecution } from "metabase-types/api";
+import type { ActionClickObject } from "./types";
+
+import { prepareParameter } from "./utils";
+
+function ActionClickDrill({ clicked }: { clicked: ActionClickObject }) {
+  const { dashboard, dashcard } = clicked.extraData;
+  const { action } = dashcard;
+
+  if (!action) {
+    return [];
+  }
+
+  const parameters: ParameterMappedForActionExecution[] = [];
+  const data = getDataFromClicked(clicked);
+
+  dashcard.parameter_mappings?.forEach?.(mapping => {
+    const parameter = prepareParameter(mapping, { action, data });
+    if (parameter) {
+      parameters.push(parameter);
+    }
+  });
```

Comment:
This is a little hard to follow setting this to an empty array and conditionally pushing into it later. I know it's a little less efficient, but is this maybe a bit easier to read?


```suggestion
  const data = getDataFromClicked(clicked);

  const parameters: ParameterMappedForActionExecution[] = 
    dashcard.parameter_mappings?.map?.(mapping => (
      prepareParameter(mapping, { action, data })
    )).filter(Boolean);
```

Reply:
Sounds good to me, but apparently a change like that doesn't make TypeScript happy, see:

```typescript
const parameterMappings = dashcard.parameter_mappings || [];

const parameters: ParameterMappedForActionExecution[] = parameterMappings
  .map(mapping => prepareParameter(mapping, { action, data }))
  .filter(Boolean);
```

1. `prepareParameter` can return `undefined`
2. `map` returns `Array<ParameterMappedForActionExecution | undefined>`
3. TypeScript is not convinced that `filter(Boolean)` would make it `Array<ParameterMappedForActionExecution>`

Check out [this StackOverflow question](https://stackoverflow.com/questions/47632622/typescript-and-filter-boolean). I'm worried a fix like that can be more confusing that iterating with `forEach` and conditionally pushing into a list :( Please LMK if you see a better solution here

Reply:
@iethree if you don't mind, I'd merge this PR now and we can iterate on the implementation details here in #25257